wordpress-mobile/WordPress-Android
# [Jetpack Content Migration Flow] Add Welcome Screen

Fixes Partially #17368

Adds a Welcome Screen after WordPress to Jetpack auto login.

⛔️ Needs splitting as it became too large. Proposed plan:

- [x] 1. #17385
- [x] 2. #17386
- [x] 3. #17387
- [x] 4. #17388
- [x] 5. #17389

---

To test:

0. **Prerequisite:** Toggle the following feature flags `on` in `Wordpress/build.gradle` code:
   (it's not possible to do this from the app because we need to be logged in to switch feature flags):
   - Set `JETPACK_SHARED_LOGIN` to `true`
   - Set `JETPACK_PROVIDER_SYNC` to `true`
   - Set `JETPACK_MIGRATION_FLOW` to `true`
   - Press the `Sync Now` button that appears at the top right of the file editing view in Android Studio
1. Build & Install the WordPress app then Login
5. Build & Install the Jetpack app **using the same variant**
   (e.g. if WordPress was `wordpressWasabiDebug` , use `jetpackWasabiDebug`)
6. **Expect** to see the "Welcome to Jetpack!" screen
7. **Verify** that the UI matches the design
8. **Expect** Closing and reopening Jetpack app to **not** show the "Welcome to Jetpack" screen
9. **Expect** Logging out and closing Jetpack app app to **not** show the "Welcome to Jetpack" screen

## Regression Notes
1. Potential unintended areas of impact
   Shared login flow.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
   Manual testing, updated tests in `SharedLoginResolverTest` for case with flag disabled.

3. What automated tests I added (or what prevented me from doing so)
   None for now.

### Preview
| <img width="314" src="https://user-images.githubusercontent.com/4588074/198731662-76746410-668a-4cb0-8c0a-a6620df92494.png"> |
| - |

PR submission checklist:
